api/opencl_runtime_layer.asciidoc

```diff
@@ -2665,14 +2665,17 @@ endif::cl_khr_external_memory[]
     _image_height_ for a 2D image array or a 3D image, must be {geq} the
     image row pitch for a 1D image array, and must be a multiple of the
     image row pitch.
-  * _num_mip_levels_ must be
-ifndef::cl_khr_mipmap_image[0.]
+  * _num_mip_levels_ must be `0`, indicating that the image has a single
+ifndef::cl_khr_mipmap_image[]
+    mipmap level.
+endif::cl_khr_mipmap_image[]
 ifdef::cl_khr_mipmap_image[]
-    0 unless the {cl_khr_mipmap_image_EXT} extension is supported, in which
-    case it may be a nonzero value specifying the number of mipmap
-    levels in the image.
+    mipmap level, unless the {cl_khr_mipmap_image_EXT} extension is supported.
```

Comment:
Shouldn't that remain "0 unless..."?

Reply:
This is a little confusing with the asciidoctor ifdefs but I think I have this right - I refactored the "must be zero" part out of the ifdefs above.

The rendered text (with all extensions enabled) is now:

> _num_mip_levels_ must be `0`, indicating that the image has a single mipmap level, unless the `cl_khr_mipmap_image` extension is supported. When the `cl_khr_mipmap_image` extension is supported, _num_mip_levels_ may additionally specify the total number of mipmap levels in the image, including the base level.

Reply:
ohh I see. yeah, that's fine then.